model-engine/model_engine_server/infra/services/image_cache_service.py

```diff
@@ -78,11 +79,14 @@ def _cache_finetune_llm_images(
         vllm_image_032 = DockerImage(
             f"{infra_config().docker_repo_prefix}/{hmi_config.vllm_repository}", "0.3.2"
         )
-        latest_tag = (
-            self.docker_repository.get_latest_image_tag(hmi_config.batch_inference_vllm_repository)
-            if not CIRCLECI
-            else "fake_docker_repository_latest_image_tag"
-        )
+        latest_tag = "fake_docker_repository_latest_image_tag"
+        if not CIRCLECI:
+            try:
+                latest_tag = self.docker_repository.get_latest_image_tag(
+                    hmi_config.batch_inference_vllm_repository
+                )
+            except ResourceNotFoundError:
```

Comment:
If we wanted to be strict about clean architecture, I feel like we'd probably want to have the service not need to know about azure (which means having the docker repository bubble up an error that was not specific to AWS/Azure/etc., and have the docker repositories themselves catch the cloud-specific error and raise the non-cloud-specific error).

Also, would we want to catch any errors that AWS/boto3 might throw at us as well?